gdk-pixbuf/src/subclass/pixbuf_animation_iter.rs

```diff
@@ -0,0 +1,178 @@
+                .expect("No parent class implementation for \"on_currently_loading_frame\"");
+
+            from_glib(f(self
+                .obj()
+                .unsafe_cast_ref::<PixbufAnimationIter>()
+                .to_glib_none()
+                .0))
+        }
+    }
+
+    fn parent_advance(&self, time: Option<Duration>) -> bool {
+        unsafe {
+            let data = T::type_data();
+            let parent_class =
+                data.as_ref().parent_class() as *mut ffi::GdkPixbufAnimationIterClass;
+            let f = (*parent_class)
+                .advance
+                .expect("No parent class implementation for \"advance\"");
+
+            let time = time.map(|t| &glib::ffi::GTimeVal {
+                tv_sec: t.as_secs() as _,
+                tv_usec: t.subsec_micros() as _,
+            } as *const _);
+            from_glib(f(
+                self.obj()
+                    .unsafe_cast_ref::<PixbufAnimationIter>()
+                    .to_glib_none()
+                    .0,
+                time.unwrap_or(std::ptr::null()),
+            ))
+        }
+    }
+}
+
+unsafe impl<T: PixbufAnimationIterImpl> IsSubclassable<T> for PixbufAnimationIter {
+    fn class_init(class: &mut ::glib::Class<Self>) {
+        Self::parent_class_init::<T>(class);
+
+        let klass = class.as_mut();
+        klass.get_delay_time = Some(animation_iter_get_delay_time::<T>);
+        klass.get_pixbuf = Some(animation_iter_get_pixbuf::<T>);
+        klass.on_currently_loading_frame = Some(animation_iter_on_currently_loading_frame::<T>);
+        klass.advance = Some(animation_iter_advance::<T>);
+    }
+}
+
+unsafe extern "C" fn animation_iter_get_delay_time<T: PixbufAnimationIterImpl>(
+    ptr: *mut ffi::GdkPixbufAnimationIter,
+) -> i32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.imp();
+
+    imp.delay_time().map(|t| t.as_millis() as i32).unwrap_or(-1)
+}
+
+unsafe extern "C" fn animation_iter_get_pixbuf<T: PixbufAnimationIterImpl>(
+    ptr: *mut ffi::GdkPixbufAnimationIter,
+) -> *mut ffi::GdkPixbuf {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.imp();
+
+    imp.pixbuf().to_glib_none().0
+}
+
+unsafe extern "C" fn animation_iter_on_currently_loading_frame<T: PixbufAnimationIterImpl>(
+    ptr: *mut ffi::GdkPixbufAnimationIter,
+) -> glib::ffi::gboolean {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.imp();
+
+    imp.on_currently_loading_frame().into_glib()
+}
+
+unsafe extern "C" fn animation_iter_advance<T: PixbufAnimationIterImpl>(
+    ptr: *mut ffi::GdkPixbufAnimationIter,
+    time_ptr: *const glib::ffi::GTimeVal,
+) -> glib::ffi::gboolean {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.imp();
+
+    let time = if time_ptr.is_null() {
+        None
+    } else {
+        Some(Duration::new(
+            (*time_ptr).tv_sec.try_into().unwrap(),
+            (*time_ptr).tv_usec.try_into().unwrap(),
```

Comment:
The `Duration` constructor takes nanoseconds, `tv_usec` sounds very much like microseconds

Reply:
Oh good catch!